docs/src/main/sphinx/sql/set-session.rst

```diff
@@ -15,28 +15,41 @@ Description
 
 Set a session property value or a catalog session property.
 
+.. _session-properties-definition:
 
-Connectors can provide session properties. They are set separately for each
-catalog by prefixing them with the catalog name. There is no sharing of
-properties across catalogs, even if the catalogs use the same connector.
+Session properties
+------------------
+
+A session property is a :doc:`configuration property </admin/properties>` that
+can be temporarily modified by a user for the duration of their current
+connection to the Trino cluster. Many configuration properties have a
+corresponding session property that accepts the same values as the config
+property.
+
+A catalog session property is a connector-defined session property that is set
+on a per-catalog basis. These properties are set separately for each catalog
+by including the catalog name as a prefix.
+
+If the user disconnects or establishes another connection to the Trino cluster,
+the values of all session properties and catalog session properties are restored
+to the cluster and its catalog's existing configuration.
 
 Examples
 --------
 
-The following example sets a system session property to enable optimized hash
+The following example sets a session property to enable optimized hash
```

Comment:
revert - IMO `system session prop` and `catalog session prop` are good to differentiate. Otherwise we'll end up using just `session prop` interchangeably for both of them which leads to confusion.

Reply:
From looking through the connector and /admin/properties docs, right now the standard seems to be that we just use "session property" for system session properties, but we do appropriately say "catalog session property" when talking about those. I could say "system session property" here but then we ought to do a sweep of the docs to update that everywhere else.

Reply:
I'd prefer that to happen for the clarity it brings - but it should be a follow up IMO.

Reply:
However I also don't feel very strongly about this as long as people don't get confused.

Reply:
Reworded and restructured this section a bit to use "Session property" as the overarching term, and be explicit about the difference between system and catalog session properties. If that looks good, then we can follow up by being descriptive about system vs catalog elsewhere, and link back here as needed.